### ISSUE TITLE
QoE Metrics Reporting - Support for BufferLevel metric

### ISSUE THREAD

**Issue by dsilhavy** [5G-MAG/rt-5gms-media-stream-handler#33]
# Feature description
Buffer level status events should be exposed by the `ExoPlayerAdapter` class and included in the QoE metrics report.

# Relevant specifications and corresponding clauses
* ISO/IEC 23009-1 5th edition - clause D.4.5
* 3GPP TS 26.501 Rel-17
* 3GPP TS 26.512 Rel-17
* 3GPP TS 26.247 - clauses 10.6.2 and 10.6.3

# Example
````xml
<QoeMetric>
  <BufferLevel>
     <BufferLevelEntry t="2011-02-16T09:08:19" level="84673"/>
     <BufferLevelEntry t="2011-02-16T09:08:20" level="93874"/>
  </BufferLevel> 
</QoeMetric>
````

**Comment by dsilhavy** [comment]
Follow up question: Should the buffer level be derived once the metrics are requested or at a certain interval. For the latter, we can then return a list of all the collected bufferlevel entries since the last request.

**Comment by dsilhavy** [comment]
Thanks @rjb1000 

> Media Session Handler sends QoE metrics reports at periodic interval specified in Service Access Information containing all of the metric values sampled since the last report was sent.

This periodic interval is in place and follows the `reportingInterval` value of the `ServiceAccessInformation.clientMetricsReportingConfigurations`

> Media Stream Handler samples each metric at (currently implementation-specific) periodic interval.

This would indeed be required with the current implementaton as the buffer level is requested via the Exoplayer API and is currently not event driven like `RepSwitch` or `HttpList`.  Maybe this would also be a configuration option provided via `M8`?

**Comment by rjb1000** [comment]
> Maybe this would also be a configuration option provided via M8?

In a real world implementation, the QoE metrics sampling period is not something that would be signalled in a service catalogue feed at reference point M8 (think DVB-I).

Ideally, the M1 and M5 APIs would be extended to explicitly specify the data sampling period to the Media Session Handler. I suggest you raise an issue in 5G-MAG/Standards.

But, as a short term measure, I think that adding a configuration parameter to our test M8 feed isn't an unreasonable way of temporarily compensating for this gap in the specifications.

**Comment by dsilhavy** [comment]
Implemented in #67
